ddl/ddl.go

```diff
@@ -467,27 +476,79 @@ func (dc *ddlCtx) jobContext(jobID int64) *JobContext {
 	return NewJobContext()
 }
 
+func (dc *ddlCtx) removeBackfillCtxJobCtx(jobID int64) {
+	dc.backfillCtx.Lock()
+	delete(dc.backfillCtx.jobCtxMap, jobID)
+	dc.backfillCtx.Unlock()
+}
+
+func (dc *ddlCtx) backfillCtxJobIDs() []int64 {
+	dc.backfillCtx.Lock()
+	defer dc.backfillCtx.Unlock()
+
+	runningJobIDs := make([]int64, 0, len(dc.backfillCtx.jobCtxMap))
+	for id := range dc.backfillCtx.jobCtxMap {
+		runningJobIDs = append(runningJobIDs, id)
+	}
+	return runningJobIDs
+}
+
+func (dc *ddlCtx) setBackfillCtxJobContext(jobID int64, jobQuery string, jobType model.ActionType) (*JobContext, bool) {
+	dc.backfillCtx.Lock()
+	defer dc.backfillCtx.Unlock()
+
+	jobCtx, existent := dc.backfillCtx.jobCtxMap[jobID]
+	if !existent {
+		dc.setDDLLabelForTopSQL(jobID, jobQuery)
+		dc.setDDLSourceForDiagnosis(jobID, jobType)
+		jobCtx = dc.jobContext(jobID)
+		dc.backfillCtx.jobCtxMap[jobID] = jobCtx
+	}
+	return jobCtx, existent
+}
+
+type reorgContexts struct {
+	sync.RWMutex
```

Comment:
I suggest that all ```sync.Mutex``` or ```sync.RWMutex``` can be replaced by ```syncutil.Mutex``` or ```syncutil.RWMutex```. It can help me find the deadlock in the CI.

Reply:
This is the old logic, I will not change it in this PR, first change the new lock(lock in `d.backfillCtx`) in this PR.